src/Microsoft.ML.TensorFlow/TensorFlow/TensorflowUtils.cs

```diff
@@ -25,7 +30,92 @@ public static void Initialize()
             ImageAnalytics.Initialize();
         }
 
+        private static unsafe ISchema GetModelSchema(IExceptionContext ectx, TFGraph graph)
+        {
+            var res = new List<KeyValuePair<string, ColumnType>>();
+            var opTypeGetters = new List<MetadataUtils.MetadataGetter<DvText>>();
+            var inputOpsGetters = new List<MetadataUtils.MetadataGetter<VBuffer<DvText>>>();
+            var inputOpsLengths = new List<int>();
+            foreach (var oper in graph)
+            {
+                if (oper.NumOutputs != 1)
```

Comment:
>if (oper.NumOutputs != 1) [](start = 16, length = 25)

I think this deserve comment.

Reply:
I removed this code. I think that if a node has multiple outputs it means that the output it produces is used as input to multiple nodes, but the shape and type of this output will be the same for every node that uses it as input. In this case there is no need to skip it. @zeahmed, is my assumption correct? 

---
In reply to: [218204247](https://github.com/dotnet/machinelearning/pull/862#discussion_r218204247) [](ancestors = 218204247)

Reply:
Actually, [tf.Operation]([https://www.tensorflow.org/api_docs/python/tf/Operation](https://www.tensorflow.org/api_docs/python/tf/Operation) ) can produced multiple outputs with different types and shapes. In case of single output used by multiple nodes, I am not sure if `oper.NumOutputs > 1' in that case. Technically, recurrent layers can produced two outputs (hidden state, output) but I would need to see how the recurrent ops are implemented in graphs.

---
In reply to: [218551502](https://github.com/dotnet/machinelearning/pull/862#discussion_r218551502) [](ancestors = 218551502,218204247)